Xamarin.Forms.Core/Storyboard.cs

```diff
@@ -0,0 +1,551 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Xamarin.Forms.Xaml;
+
+namespace Xamarin.Forms
+{
+	[ContentProperty(nameof(Children))]
+	public class Storyboard : Timeline, IList<Timeline>
+	{
+		Dictionary<VisualElement, List<Animation>> _animationsPerElement = new Dictionary<VisualElement, List<Animation>>();
+		List<Animation> _runningAnimations = new List<Animation>();
+
+
+		List<Timeline> _timelines;
+
+		public List<Timeline> Children => _timelines;
+
+		public int Count => Children.Count;
+
+		public bool IsReadOnly => false;
+
+		public Timeline this[int index] { get => Children[index]; set => Children[index] = value; }
+
+		public static readonly BindableProperty TargetNameProperty = BindableProperty.CreateAttached("TargetName", typeof(VisualElement), typeof(Timeline), null,
+			propertyChanged: (bindable, oldValue, newValue) =>
+			{
+				foreach (var item in (bindable as Storyboard)?.Children)
+				{
+					item.SetValue(BindingContextProperty, (newValue as VisualElement).BindingContext);
+				}
+			});
+
+		public static readonly BindableProperty TargetPropertyProperty = BindableProperty.CreateAttached("TargetProperty", typeof(BindableProperty), typeof(Timeline), null);
+
+		public static void SetTarget(BindableObject bindable, VisualElement value)
+		{
+			bindable.SetValue(TargetNameProperty, value);
+			SetInheritedBindingContext(bindable, value?.BindingContext);
+		}
+
+		public static void SetTargetName(BindableObject bindable, string value)
+		{
+			bindable.SetValue(TargetNameProperty, value);
+		}
+
+		public static VisualElement GetTargetName(BindableObject bindable)
+		{
+			return (VisualElement)bindable.GetValue(TargetNameProperty);
+		}
+
+		public static VisualElement GetTarget(BindableObject bindable)
+		{
+			return (VisualElement)bindable.GetValue(TargetNameProperty);
+		}
+
+		public static void SetTargetProperty(BindableObject bindable, BindableProperty value)
+		{
+			bindable.SetValue(TargetPropertyProperty, value);
+		}
+
+		protected override void OnBindingContextChanged()
+		{
+			base.OnBindingContextChanged();
+			foreach (var item in Children)
+			{
+				item.SetValue(BindingContextProperty, BindingContext);
+				//SetInheritedBindingContext(item, BindingContext);
```

Comment:
There's several code commented of lots of empty new lines...

Reply:
just me hacking :)